conans/client/loader.py

```diff
@@ -16,6 +16,180 @@
+        # Put all imported files under a new package name
+        module_id = uuid.uuid1()
+        added_modules = set(sys.modules).difference(old_modules)
+        for added in added_modules:
+            module = sys.modules[added]
+            if module:
+                folder = os.path.dirname(module.__file__)
+                if folder.startswith(current_dir):
+                    module = sys.modules.pop(added)
+                    sys.modules["%s.%s" % (module_id, added)] = module
+    except Exception:
+        import traceback
+        trace = traceback.format_exc().split('\n')
+        raise ConanException("Unable to load conanfile in %s\n%s" % (conan_file_path,
+                                                                     '\n'.join(trace[3:])))
+    finally:
+        sys.path.pop()
+
+    return loaded, filename
+
+
+def load_conanfile_class(conanfile_path, check_name_version=False):
+    """ Load only the class of the ConanFile recipe, but do not instantiate the object
+    It is needed for the 'conan export' command
+    """
+    loaded, filename = _parse_file(conanfile_path)
+    try:
+        result = _parse_module(loaded, filename)
+        # Exports is the only object field, we need to do this, because conan export needs it
+        result.exports = create_exports(result)
+        result.exports_sources = create_exports_sources(result)
+    except Exception as e:  # re-raise with file name
+        raise ConanException("%s: %s" % (conanfile_path, str(e)))
+
+    # check name and version were specified
+    if check_name_version:
+        if not hasattr(result, "name") or not result.name:
+            raise ConanException("conanfile didn't specify name")
+        if not hasattr(result, "version") or not result.version:
+            raise ConanException("conanfile didn't specify version")
+
+    return result
+
+
+def _loader(runner, settings, profile):
+    return ConanFileLoader(runner,
+                           settings=settings,
+                           package_settings=profile.package_settings_values,
+                           options=profile.options, scopes=profile.scopes,
+                           env_values=profile.env_values)
+
+
+def _get_single_loader(current_path, settings, runner):
+    profile = Profile()
+    conan_info_path = os.path.join(current_path, CONANINFO)
+    if conan_info_path and os.path.exists(conan_info_path):
+        existing_info = ConanInfo.load_file(conan_info_path)
+        settings.values = existing_info.full_settings
+        profile.options = existing_info.full_options
+        profile.scopes = existing_info.scope
+        profile.env_values = existing_info.env_values
+
+    return _loader(runner, settings, profile)
+
+
+def install_loader(settings, profile, runner):
+    settings.values = profile.settings_values
+    return _loader(runner, settings, profile)
+
+
+def load_conanfile_single(conanfile_path, current_path, settings, runner, output, reference=None,
```

Comment:
XDDD
It means it will only load a single, unique, 1 conanfile. For consumer commands (build, package, source). The opposite is the "install_loader" used for really building the full deps graph and then loading many conanfiles.